manager/src/main/java/com/redhat/service/bridge/manager/api/user/validators/templates/ValidTransformationTemplate.java

```diff
@@ -0,0 +1,21 @@
+package com.redhat.service.bridge.manager.api.user.validators.templates;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Target({ ElementType.PARAMETER })
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = TransformationTemplateValidatorContainer.class)
+public @interface ValidTransformationTemplate {
```

Comment:
@sutaakar I'm wondering if this is better placed as a `FIELD` level annotation on the `ProcessorRequest` object so it would be used like:

```
public class ProcessorRequest {
  @ValidTransformationTemplate
  @JsonProperty("transformationTemplate")
  private String transformationTemplate;
}
```

From what I can see, the validation doesn't need access to any other part of the `ProcessorRequest` entity other than the `transformationTemplate` field.

We would then do the same with the `@ValidActionParams` annotation so that our controller ends up looking like this:

```
public Response addProcessorToBridge(@PathParam("bridgeId") @NotEmpty String bridgeId, @Valid ProcessorRequest processorRequest)
```
and the validation requirements are completely encapsulated within the `ProcessorRequest` entity.

For the sake of consistency, I suggest we merge your PR in its current form and then put an issue in the backlog to review placement of our validation annotations.

We also have https://issues.redhat.com/browse/MGDOBR-202 to consider that was created by @lucamolteni 

WDYT?

Reply:
Sounds fine for me, will create a JIRA to address it

Reply:
@robpblake Just created as https://issues.redhat.com/browse/MGDOBR-491

Reply:
Perfect. Thanks @sutaakar 🚀